File: leetcode/easy/1108/1108.rb
=begin
Runtime: 44ms, faster than 83.48%
Memory Usage: 209.8 MB
=end
def defang_i_paddr(address)
    i = 1
    while (i < address.length) do
        if address[i-1] == '.'
            address.insert(i-1, '[')
            address.insert(i+1, ']')
            i += 2
        end
        i += 1
    end
    address
end

File: easy/1108/1108.cpp
/**
 * Runtime: 0 ms, faster than 100.00%
 * Memory Usage: 6.1 MB, less than 86.04%
 */
class Solution {
public:
    string defangIPaddr(string address) {
        string s = "";
        for(int i=0; address[i]; ++i)
        {
            if(address[i] != '.') s += address[i];
            else 
            {
                s += "[.]";
                
            }
        }
        
        return s;
    }
};
